source/neuropod/bindings/java/src/main/java/com/uber/neuropod/NeuropodTensor.java

```diff
@@ -0,0 +1,146 @@
+package com.uber.neuropod;
+
+import java.io.Serializable;
+import java.nio.*;
+import java.util.List;
+
+/**
+ * This class is used for holding tensor data. It has its underlying C++ NeuropodTensor
+ * object, should call close() function to free C++ side object when finish using the
+ * NeuropodTensor object.
+ * This object can be created by the Java side or by the C++ side(as an inference result).
+ */
+public class NeuropodTensor extends NativeClass implements Serializable {
+     /**
+      * Get the dims array which represnents the shape of a tensor.
+      *
+      * @return the shape array
+      */
+     public long[] getDims() {return null;}
+
+     /**
+      * Gets the number of elements of a tensor.
+      *
+      * @return the number of elements
+      */
+     public long getNumberOfElements() {return 0;}
+
+     /**
+      * Gets the type of a tensor
+      *
+      * @return the tensor type
+      */
+     public TensorType getTensorType() {return null;}
+
+     /**
+      * Flatten the tensor data and convert it to a long buffer.
+      * <p>
+      * Can only be used when the tensor is INT64_TENSOR. Will trigger
+      * a copy.
+      *
+      * @return an array backed LongBuffer
```

Comment:
Do we need these to be array backed buffers?

Reply:
Yeah, this way users can know they can call array() function to get the array form of the data.

Reply:
A user can just call [hasArray](https://docs.oracle.com/javase/7/docs/api/java/nio/LongBuffer.html#hasArray()) to see if they can call `array()`.

I don't think we want to commit to always copying the data just because a user could call `array()`. Since we provide elementwise access (below), I think it's okay for us to return non array-backed buffers here.

Users can also do this copy on their end so they're only paying the performance penalty of the features they're using.

Reply:
I see. I've deleted the an array backed part.